crypto/evp/evp_lib.c

```diff
@@ -242,6 +242,13 @@ int EVP_CIPHER_iv_length(const EVP_CIPHER *cipher)
 
 int EVP_CIPHER_CTX_iv_length(const EVP_CIPHER_CTX *ctx)
 {
+    int i, rv;
+
+    if ((EVP_CIPHER_flags(ctx->cipher) & EVP_CIPH_CUSTOM_IV_LENGTH) != 0) {
+        rv = EVP_CIPHER_CTX_ctrl((EVP_CIPHER_CTX *)ctx, EVP_CTRL_GET_IVLEN,
+                                 0, &i);
+        return (rv == 1) ? i : -1;
```

Comment:
If `EVP_CIPHER_CTX_ctrl()` returns `EVP_CTRL_RET_UNSUPPORTED` (i.e. -1, as opposed to 0 for error), shouldn't that have this function fall back to return `ctx->cipher->iv_len`?  That's what other functions of this kind does.

In other words, this return line could be replaced with this:

``` C
        if (rv >= 0)
            return (rv != 0) ? i : -1;
```

Reply:
Ah, you're going to claim that the flag means `EVP_CIPHER_CTX_ctrl()` never returns -1, is that it?

Reply:
Yep - the flag means I handle this ctrl - if it fails in any way then its an error

Reply:
That kind of changes the semantics and is one extra complexity level, but [shrug] ok...